src/client/main.ts

```diff
@@ -420,24 +420,53 @@ function deserializeChannelChangedEvent(eventTransport: Transport<ChannelChanged
 
 if (typeof fin !== 'undefined') {
     getServicePromise().then((channelClient) => {
-        channelClient.register(APIToClientTopic.RECEIVE_INTENT, (payload: RaiseIntentPayload) => {
-            intentListeners.forEach((listener: IntentListener) => {
-                if (payload.intent === listener.intent) {
-                    listener.handler(payload.context);
+        channelClient.register(APIToClientTopic.RECEIVE_INTENT, async (payload: RaiseIntentPayload) => {
+            let successes = 0;
+            let failures = 0;
+
+            await Promise.all(intentListeners.filter((listener) => payload.intent === listener.intent).map(async (listener) => {
+                try {
+                    await listener.handler(payload.context);
+                    successes++;
+                } catch (e) {
+                    failures++;
+                    console.warn(`Error thrown by ${payload.intent} intent handler, swallowing error. Error message: ${e.message}`);
                 }
-            });
+            }));
+
+            if (failures > 0 && successes === 0) {
+                throw new Error(`All ${payload.intent} intent handlers failed`);
+            }
         });
 
         channelClient.register(APIToClientTopic.RECEIVE_CONTEXT, (payload: ReceiveContextPayload) => {
+            let successes = 0;
+            let failures = 0;
```

Comment:
Can we avoid the duplication here? (and in contextChannels.ts too)

Should the `collateApiCallResults` util be in `client/internal.ts`, much like the error code utils?

Reply:
Removed the client side duplication, but still distinct from the "collate" call on the provider side. Think they are sufficiently different for it to not make sense to unify the two

✅